### PR TITLE
Add Kling-Gupta Efficiency and differentiate between modified an unmodified

### PR DESCRIPTION
**Pull request by martinvonk**
# Short Description
See title.

Also deprecates kge_2012 in version 2.0

# Checklist before PR can be merged:
- [x] closes issue #606
- [x] is documented
- [x] Format code with [Black formatting](https://black.readthedocs.io)
- [x] type hints for functions and methods
- [x] tests added / passed

### CODE REVIEW

File: pastas/modelstats.py [hunk 1]
```diff
@@ -278,6 +279,36 @@ def rsq(
...
+
+        Parameters
+        ----------
+        tmin: str or pandas.Timestamp, optional
+        tmax: str or pandas.Timestamp, optional
+        weighted: bool, optional
+            If weighted is True, the variances are computed using the time step
+            between observations as weights. Default is False.
+        modified: bool, optional
+            Use the modified kge if True
```
Comment (raoulcollenteur) [inline pastas/modelstats.py:301]: This needs a reference.. no one know what modified means

File: pastas/stats/metrics.py [hunk 2]
```diff
@@ -536,6 +542,8 @@ def kge_2012(
         maximum allowed gap period in days to use for the computation of the
         weights. All time steps larger than max_gap are replace with the
         max_gap value. Default value is 30 days.
+    modified: bool, optional
```
Comment (raoulcollenteur) [inline pastas/stats/metrics.py:545]: Add reference

File: tests/test_metrics.py
Comment (raoulcollenteur) [inline tests/test_metrics.py:1]: Co-pilot? :)

**Review by raoulcollenteur — APPROVED** [review]
Add references, otherwise fine with me

**Comment by raoulcollenteur** [comment]
Hi @martinvonk, can you add the reference? Otherwise I think this is fine, i like the change to `kge`.

Cheers,
Raoul

**Comment by martinvonk** [comment]
Can you add me to the Zotero so I can add the reference?

**Comment by martinvonk** [comment]
Reference was in the notes but is now also present in the parametes list:
```
modified: bool, optional
    Use the modified KGE as proposed by :cite:t:`kling_runoff_2012`.
    According to the article this ensures that the bias and variability
    ratios are not cross-correlated, which otherwise may occur when inputs
    are biased.

Notes
-----
The (weighted) Kling-Gupta Efficiency :cite:t:`kling_runoff_2012` is
computed as follows:

.. math:: \\text{KGE} = 1 - \\sqrt{(r-1)^2 + (\\beta-1)^2 - (\\gamma-1)^2}

where :math:`\\beta = \\bar{x} / \\bar{y}` and :math:`\\gamma =
\\frac{\\bar{\\sigma}_x}{\\bar{\\sigma}_y}}`. If modified equals True,
:math:`\\gamma = \\frac{\\bar{\\sigma}_x / \\bar{x}}{\\bar{\\sigma}_y /
\\bar{y}}`. If weighted equals True, the weighted mean, variance and
pearson correlation are used.
```

**Comment by martinvonk** [comment]
Do we have a list somewhere where everything is listed which we intend to deprecate for Pastas 2.0?

**Pull request merged**